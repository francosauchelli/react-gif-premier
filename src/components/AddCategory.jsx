import { useState } from "react"

import Proptypes from 'prop-types';


export const AddCategory = ({ onNewCategory }) => {

    const [ inputValue, setInputValue ] = useState( '' );

    const onInputChange  = ({ target }) => {
        setInputValue( target.value );
    }

    const onSubmit = ( event ) => {
        event.preventDefault();

        const newValue = inputValue.trim();
        if( newValue.length <= 1 ) return;

        onNewCategory( newValue );
        setInputValue('');
    }

    return (
        <form 
            onSubmit={ onSubmit } 
            aria-label='form' 
        >
            <input 
                type="text"
                placeholder="Buscar Gifs"
                onChange={ onInputChange }
                value={ inputValue } 
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: Proptypes.func.isRequired,
}