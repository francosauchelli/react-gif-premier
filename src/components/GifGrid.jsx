import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {

    const { gifs, isLoading } = useFetchGifs( category );


    return (
        <>  
            {
                isLoading && ( <h2>Loading...</h2> )
            }
            <h3>{ category }</h3>
            <div className='card-grid'>

                { gifs && gifs.map( ( gif ) => 
                    <GifItem key={ gif.id }
                        { ...gif } 
                    /> 
                )}
            </div>        
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}