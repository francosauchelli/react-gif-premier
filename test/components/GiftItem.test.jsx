import { render, screen } from "@testing-library/react";

import { GifItem } from "../../src/components/GifItem";


describe( 'Prueba <GiftItem />', () => {

    const title = 'Gif Title';
    const urlImage = 'https://url.com/img.jpg'


    test( 'debe hacer match el snapshot', () => {

        const { container } = render( <GifItem title={ title } image={ urlImage } /> );

        expect( container ).toMatchSnapshot();
    });

    test( 'debe mostrar imagen con el url y alt indicados', () => {
        render( <GifItem title={ title } image={ urlImage } /> );

        const { src, alt } = screen.getByRole( 'img' );

        expect( src ).toBe( urlImage );
        expect( alt ).toBe( title );

    });

    test( 'Debe existir el tÃ­tulo en el componente', () => {
        render( <GifItem title={ title } image={ urlImage } /> );

        expect( screen.getByText( title ) ).toBeTruthy();
    })
})