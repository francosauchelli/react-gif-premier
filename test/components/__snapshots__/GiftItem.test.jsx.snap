// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <GiftItem /> debe hacer match el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Gif Title"
      src="https://url.com/img.jpg"
    />
    <p>
      Gif Title
    </p>
  </div>
</div>
`;
